src/ImageSharp/Formats/Tiff/TiffFrameMetadata.cs

```diff
@@ -0,0 +1,370 @@
+
+namespace SixLabors.ImageSharp.Formats.Experimental.Tiff
+{
+    /// <summary>
+    /// Provides Tiff specific metadata information for the frame.
+    /// </summary>
+    public class TiffFrameMetadata : IDeepCloneable
+    {
+        // 2 (Inch)
+        internal const ushort DefaultResolutionUnit = 2;
+
+        private const TiffPlanarConfiguration DefaultPlanarConfiguration = TiffPlanarConfiguration.Chunky;
+
+        private const TiffPredictor DefaultPredictor = TiffPredictor.None;
+
+        private ExifProfile frameTags;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TiffFrameMetadata"/> class.
+        /// </summary>
+        public TiffFrameMetadata()
+        {
+        }
+
+        /// <summary>
+        /// Gets the Tiff directory tags.
+        /// </summary>
+        public ExifProfile ExifProfile
+        {
+            get => this.frameTags ??= new ExifProfile();
+            internal set => this.frameTags = value;
+        }
+
+        /// <summary>
+        /// Gets a general indication of the kind of data contained in this subfile.
+        /// </summary>
+        /// <value>A general indication of the kind of data contained in this subfile.</value>
+        public TiffNewSubfileType SubfileType => (TiffNewSubfileType?)this.ExifProfile.GetValue(ExifTag.SubfileType)?.Value ?? TiffNewSubfileType.FullImage;
+
+        /// <summary>
+        /// Gets a general indication of the kind of data contained in this subfile.
+        /// </summary>
+        /// <value>A general indication of the kind of data contained in this subfile.</value>
+        public TiffSubfileType? OldSubfileType => (TiffSubfileType?)this.ExifProfile.GetValue(ExifTag.OldSubfileType)?.Value;
+
+        /// <summary>
+        /// Gets the number of columns in the image, i.e., the number of pixels per row.
+        /// </summary>
+        public Number Width
+        {
+            get
+            {
+                IExifValue<Number> width = this.ExifProfile.GetValue(ExifTag.ImageWidth);
+                if (width == null)
+                {
+                    TiffThrowHelper.ThrowImageFormatException("The TIFF image is missing the ImageWidth");
+                }
+
+                return width.Value;
+            }
+        }
+
+        /// <summary>
+        /// Gets the number of rows of pixels in the image.
+        /// </summary>
+        public Number Height
+        {
+            get
+            {
+                IExifValue<Number> height = this.ExifProfile.GetValue(ExifTag.ImageLength);
+                if (height == null)
+                {
+                    TiffThrowHelper.ThrowImageFormatException("The TIFF image is missing the ImageLength");
+                }
+
+                return height.Value;
+            }
+        }
+
+        /// <summary>
+        /// Gets the number of bits per component.
+        /// </summary>
+        public ushort[] BitsPerSample
```

Comment:
Why is this an array? The bit depth isn't variable in a single frame is it?

Reply:
@JimBobSquarePants In the `Tiff` format, they are stored like this. For sample case `RGB 888` will be `{8, 8, 8}` and `PhotometricInterpretation == RGB`, but there are can store variable values, for example famous `{5, 6, 5}, {5, 5, 5}` and `RGB`

Reply:
Then I would suggest we map that to an enum for public consumption. An array doesn’t provide a safe enumeration of allowable values

Reply:
This is now an enum with 80771720